manager/apis/app/v1alpha1/m4dapplication_types.go

```diff
@@ -42,10 +42,8 @@ type DataRequirements struct {
 	Copy CopyRequirements `json:"copy,omitempty"`
 }
 
-// DataContext indicates data set chosen by the Data Scientist to be used by his application,
-// and includes information about the data format and technologies used by the application
-// to access the data.
-type DataContext struct {
+// AssetDetails provides all details required to fully identify an asset - the datasetID and the catalog details
+type AssetDetails struct {
```

Comment:
Catalog service does not identify the asset, merely suggests what service to refer to for the lookup. 
I prefer you leave the current spec because the catalog service is not used yet, and adding an hierarchy only complicates the spec. There is an additional cost to changing the spec: in all samples we have, documentation, plus changing the code of GUI backend/frontend s.t. the GUI functionality is preserved.